integration/cdi/src/main/java/org/apache/cxf/cdi/CdiClassUnwrapper.java

```diff
@@ -22,8 +22,18 @@
 
 import org.apache.cxf.common.util.ClassUnwrapper;
 
+/**
+ * Unwraps the CDI proxy classes into real classes.
+ */
 class CdiClassUnwrapper implements ClassUnwrapper {
-    private static final Pattern PROXY_PATTERN = Pattern.compile(".+\\$\\$.+Proxy");
+    /**
+     * Known proxy patterns for OWB and Weld:
+     * 
+     *  Xxx$$OwbNormalScopeProxy0
+     *  Xxx$Proxy$_$$_WeldClientProxy
+     *  
+     */
+    private static final Pattern PROXY_PATTERN = Pattern.compile(".+\\$\\$.+Proxy\\d*");
```

Comment:
are you sure its always digits on the end?

Reply:
I see it consistently on OWB but it may not be always (hereby `\d*` to make this part optional)

Reply:
should it perhaps then be `.*`

Reply:
Hm ... not sure,`\d*` could be explained (and I have example for it), .* not sure ....